docs/resources/function.md

```diff
@@ -0,0 +1,139 @@
+---
+subcategory: "Unity Catalog"
+---
+# databricks_function Resource
+
+-> This resource source can only be used with a workspace-level provider.
+
+Creates a [User-Defined Function (UDF)](https://docs.databricks.com/en/udf/unity-catalog.html) in Unity Catalog. UDFs can be defined using SQL, or external languages (e.g., Python) and are stored within [Unity Catalog schemas](../resources/schema.md). 
+
+## Example Usage
+
+### SQL-based function:
+
+```hcl
+resource "databricks_catalog" "sandbox" {
+  name    = "sandbox_example"
+  comment = "Catalog managed by Terraform"
+}
+
+resource "databricks_schema" "functions" {
+  catalog_name = databricks_catalog.sandbox.name
+  name         = "functions_example"
+  comment      = "Schema managed by Terraform"
+}
+
+resource "databricks_function" "calculate_bmi" {
+  name              = "calculate_bmi"
+  catalog_name      = databricks_catalog.sandbox.name
+  schema_name       = databricks_schema.functions.name
+  input_params      = [
+    {
+      name = "weight"
+      type = "DOUBLE"
+    },
+    {
+      name = "height"
+      type = "DOUBLE"
+    }
+  ]
+  data_type         = "DOUBLE"
+  routine_body      = "SQL"
+  routine_definition = "weight / (height * height)"
+  language          = "SQL"
+  is_deterministic  = true
+  sql_data_access   = "CONTAINS_SQL"
+  security_type     = "DEFINER"
+}
```

Comment:
Based on my reading of the FunctionInfo type, this will need to look a bit different. Not your fault, as some of these things have changed since you started writing this PR. Specifically, the `input_params` is an object with a list nested object field `parameters`.

```suggestion
resource "databricks_function" "calculate_bmi" {
  name              = "calculate_bmi"
  catalog_name      = databricks_catalog.sandbox.name
  schema_name       = databricks_schema.functions.name
  input_params      = {
    parameters = [{
      name = "weight"
      type_name = "DOUBLE"
    },
    {
      name = "height"
      type_name = "DOUBLE"
    }]
  ]
  data_type         = "DOUBLE"
  routine_body      = "SQL"
  routine_definition = "weight / (height * height)"
  is_deterministic  = true
  sql_data_access   = "CONTAINS_SQL"
  security_type     = "DEFINER"
}
```